cfme/tests/cloud/test_cloud_events.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+"""This module tests only cloud specific events"""
+import pytest
+import re
+
+from cfme.common.vm import VM
+from utils import testgen
+from utils.events import EventBuilder
+from utils.generators import random_vm_name
+
+
+pytest_generate_tests = testgen.generate(testgen.cloud_providers, scope="module")
```

Comment:
Why generate on all cloud providers if all tests only collect for azure?

Consider testgen.provider_by_type(['azure'])